packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2029,8 +2027,8 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
       fillColor: _getFillColor(themeData),
       focusColor: _getFocusColor(themeData),
       hoverColor: _getHoverColor(themeData),
-      isFocused: isFocused,
-      isHovering: isHovering,
+      isFocused: isFocused && decoration.enabled,
```

Comment:
Is there anywhere that we actually could have `isFocused` be true when `decoration.enabled` is false? If not, maybe we should add this logic to the `isFocused` getter?

```dart
bool get isFocused => widget.isFocused && decoration.enabled;
```

Reply:
Same question for `isHovering`

Reply:
Great idea.  Done.